### PR TITLE
REF: de-duplicate code in libparsing/libperiod

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_libs/tslibs/parsing.pyx [hunk 1]
```diff
@@ -438,21 +438,13 @@ cdef inline object _parse_dateabbr_string(object date_string, datetime default,
...
-                                         f'information from given '
-                                         f'freq: {freq}')
-
-                month = (mnum + (quarter - 1) * 3) % 12 + 1
-                if month > mnum:
-                    year -= 1
-            else:
-                month = (quarter - 1) * 3 + 1
+            try:
+                # TODO: hack attack, GH#1228
```
Comment (jreback) [inline pandas/_libs/tslibs/parsing.pyx:442]: prob can remove the comment now :->

File: pandas/_libs/tslibs/period.pyx [hunk 2]
```diff
@@ -74,8 +74,7 @@ from pandas._libs.tslibs.dtypes cimport (
     PeriodDtypeBase,
     attrname_to_abbrevs,
 )
-from pandas._libs.tslibs.parsing cimport get_rule_month
-
+from pandas._libs.tslibs.parsing cimport quarter_to_myear
```
Comment (jreback) [inline pandas/_libs/tslibs/period.pyx:77]: prob can cimport parse_time_string too

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. small comments

File: pandas/_libs/tslibs/parsing.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/parsing.pyx:442]: updated

File: pandas/_libs/tslibs/period.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/period.pyx:77]: its not cpdef

**Review by jreback — APPROVED** [review]

**Pull request merged**